test/check/check.go

```diff
@@ -353,6 +353,26 @@ func checkForDeployment(name string) error {
 	return nil
 }
 
+// CheckOperatorPodStability makes sure that the CNAO pod has not restarted since it started working
+func CheckOperatorPodStability() {
+	pods := corev1.PodList{}
+	listOptions := client.ListOptions{}
+	labelSelector, err := k8slabels.Parse(fmt.Sprintf("name, name == %s", components.Name))
+	Expect(err).NotTo(HaveOccurred(), "should succeed parsing label selector")
+	listOptions.LabelSelector = labelSelector
+	listOptions.Namespace = components.Namespace
+	err = framework.Global.Client.List(context.Background(), &pods, &listOptions)
+	Expect(err).NotTo(HaveOccurred(), "should succeed getting the cnao pod")
+
+	Expect(len(pods.Items)).To(BeNumerically("==", 1), fmt.Sprintf("should only have 1 cnao pod"))
```

Comment:
If we know that there is only one pod, why not just get  the CNAO pod,  instead of get a list of pods?

Reply:
Done

Reply:
We don't have the pod name since it's generated so we cannot use get on it we have to list and filter.

Reply:
but Alon is right in the manner that we don't need to iterate over a pod list with len 1. fixed below